src/main/java/com/google/sps/data/Event.java

```diff
@@ -0,0 +1,184 @@
+ 
+  // class constants
+  private final static int HALFHOUR = 30;
+  private final static int HOUR = 60;
+  private final static int MINUTES_IN_A_DAY = 1440;
+  private final static int MIN_POSSIBLE_TIME = 0;
+
+  /**
+   * Constructor that takes in time spent at location
+   * 
+   * @param name date from user input travel day. Cannot be null.
+   * @param address time to to start or end event. Must be within bounds
+   * @param date day of travel
+   * @param startTime start of activing
+   * @param travelTime time spent traveling to next location
+   * @param timeAtLocation time spent at POI
+   */
+  public Event(String name, String address, String date, int startTime, 
+              int travelTime, int timeAtLocation) {
+    this.name = name;
+    this.address = address;
+    this.date = date;
+    this.startTime = startTime;
+    this.travelTime = travelTime;
+    this.endTime = calculateEndTime(timeAtLocation);
+
+    createCalendarTimes();
+  }
+ 
+  /**
+   * Constructor that assume one hour default spent at location
+   * 
+   * @param name date from user input travel day. Cannot be null.
+   * @param address time to to start or end event. Must be within bounds
+   * @param date day of travel
+   * @param startTime start of activing
+   * @param travelTime time spent traveling to next location
+   */
+  public Event(String name, String address, String date, int startTime, int travelTime) {
+    this.name = name;
+    this.address = address;
+    this.date = date;
+    this.startTime = startTime;
+    this.travelTime = travelTime;
+    this.endTime = calculateEndTime(HOUR);
+
+    createCalendarTimes();
+  }
+
+  // function that sets start and end Time with correct string format
+  private void createCalendarTimes() {
+    this.strStartTime = createStrTime(this.date, this.startTime);
+    this.strEndTime = createStrTime(this.date, this.endTime);
+  }
+
+  /**
+   * function that builds string w correct format for calendar-script.js
+   * 
+   * @param date date from user input travel day. Cannot be null.
+   * @param time time to to start or end event. Must be within bounds
+   */
+  private static String createStrTime(String date, int time) {
+    if (date == null) {
+      throw new NullPointerException("Date cannot be null");
+    }
+
+    if (time < MIN_POSSIBLE_TIME) {
+      throw new IllegalArgumentException("Time cannot be less than 0");
+    }
+
+    if (time >= MINUTES_IN_A_DAY) {
+      throw new IllegalArgumentException("Time cannot be more than a 24hrs");
+    }
+
+    String output = "";
+    output += date + "T" + Integer.toString(time).substring(0, 2) + ":" + 
+      Integer.toString(time).substring(2, 4) + ":00";
+    return output;
+  }
+
+  /**
+   * function that calculates endTime given start and timeSpent
+   * 
+   * @param timeAtLocation time that the user spends at a POI
+   */
+  private int calculateEndTime(int timeAtLocation) {
+    return this.startTime + Integer.parseInt(convertToFormat(timeAtLocation));
```

Comment:
Would this create issues if the minutes exceeds 60? For example if the startTime is 3:45 and timeAtLocation is 30 mins wouldn't this return 3:75? Just want to clarify what values you're adding here

Reply:
you are right